src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE = "Index out of bounds: ";
+    private static final String ELEMENT_NOT_FOUND_MESSAGE = "Element not found: ";
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_MULTIPLIER = 1.5;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(Object[] elementData) {
+        this.elementData = new Object[elementData.length];
+        System.arraycopy(elementData, 0, this.elementData, 0, elementData.length);
+        this.size = elementData.length;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_OUT_OF_BOUNDS_MESSAGE + index);
+        }
+        if (size == elementData.length) {
+            elementData = grow();
```

Comment:
![Screenshot (126)](https://github.com/mate-academy/jv-array-list/assets/112598705/deab2655-15c9-4a68-9f06-6dae2783b575)
i guess we can remove that if block and make that condition in grow() method